module-services/service-desktop/endpoints/include/endpoints/deviceInfo/DeviceInfoEndpointCommon.hpp

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include <endpoints/Endpoint.hpp>
+#include <Service/Service.hpp>
+
+#include <string>
+#include <tuple>
+
+#include <json11.hpp>
+
+namespace sdesktop::endpoints
+{
+    class DeviceInfoEndpointCommon : public Endpoint
+    {
+      public:
+        enum class DiagnosticFileType
+        {
+            Logs,
+            CrashDumps
+        };
+
+        auto handleGet(Context &context) -> http::Code;
+        auto gatherListOfDiagnostics(Context &context, DiagnosticFileType diagDataType) -> http::Code;
+        auto parseDiagnosticFileType(const json11::Json &fileList) -> DiagnosticFileType;
+        auto listDirectory(std::string path) -> std::vector<std::string>;
+        auto fileListToJsonObject(const std::vector<std::string> &fileList) const -> json11::Json::object const;
+        auto requestLogsFlush() const -> void;
+        auto getStorageStats(const std::string &path) -> std::tuple<long, long>;
+        auto getStorageInfo() -> std::tuple<long, long, long>;
+        auto getDeviceToken() -> std::string;
+
+        static constexpr auto OS_RESERVED_SPACE_IN_MB = (1024LU);
+
+      public:
```

Comment:
Why `public` access specifier is added two times here?

Reply:
Fixed